plugins/woocommerce-blocks/assets/js/blocks/cart/inner-blocks/cart-order-summary-subtotal/block.tsx

```diff
@@ -5,13 +5,29 @@ import { Subtotal, TotalsWrapper } from '@woocommerce/blocks-components';
 import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
 import { useStoreCart } from '@woocommerce/base-context/hooks';
 
-const Block = ( { className = '' }: { className?: string } ): JSX.Element => {
+export type BlockAttributes = {
+	className: string;
```

Comment:
```suggestion
	className?: string | undefined;
```

Shouldn't this definition also check against an optional element that can be `undefined`?

Reply:
I don't think it needs to be `className?: string | undefined`

But you're right that it's probably not string. I copied the type def from here: https://github.com/woocommerce/woocommerce/pull/53544/files/323cc868b3139fdea19f33153c43cc8656380552#diff-090b2440beb63e4a22de7aac648a50ee7cb9cc3023d39e4c9033cca4dab09e60L14-L16 because that's whats passed down.

I have actually seen className typed many ways throughout the code base.

in JSX it's a little nuanced:

lets say that the `Edit` block has `undefined` for the `className` attribute like [here](https://github.com/woocommerce/woocommerce/pull/53544/files/323cc868b3139fdea19f33153c43cc8656380552#diff-090b2440beb63e4a22de7aac648a50ee7cb9cc3023d39e4c9033cca4dab09e60R18) and we always render the Block like: `<Block className={className} ... />` like [here](https://github.com/woocommerce/woocommerce/pull/53544/files/323cc868b3139fdea19f33153c43cc8656380552#diff-090b2440beb63e4a22de7aac648a50ee7cb9cc3023d39e4c9033cca4dab09e60R41) then using `?` is incorrect, the property will always be present, but assuming the `className` in edit could be undefined then the correct type would be `className: string | undefined`

That's because we always pass the property key by doing `className={ className }`. if we did something different like (naive example

```
let className: string | undefined;

const props = {
  foo: 'bar'
}

if ( className ) { props.className = className }
 
 <Block {...props} /> 
 ```
then the type could indeed be `className?: string | undefined`

Reply:
For a while now I'd been asking myself, actually is this className type correct at all, we tend to type it incorrectly and inconsistently, but I just got to the bottom of it.

`BlockEditProps` is a convenience type for the props passed to edit. it extends `BlockSaveProps` which has `className` typed as string. Which makes sense. Actually these **should all be string**. In that case I would assume that the frontend block should also have className as `string` too.

Reply:
> lets say that the Edit block has undefined for the className attribute like [here](https://github.com/woocommerce/woocommerce/pull/53544/files/323cc868b3139fdea19f33153c43cc8656380552#diff-090b2440beb63e4a22de7aac648a50ee7cb9cc3023d39e4c9033cca4dab09e60R18) and we always render the Block like: <Block className={className} ... /> like [here](https://github.com/woocommerce/woocommerce/pull/53544/files/323cc868b3139fdea19f33153c43cc8656380552#diff-090b2440beb63e4a22de7aac648a50ee7cb9cc3023d39e4c9033cca4dab09e60R41) then using ? is incorrect, the property will always be present, but assuming the className in edit could be undefined then the correct type would be className: string | undefined

That's what I looked up. In `edit.tsx`, I noticed that `className` is `undefined`.

Reply:
@nielslange Yep I was pointing out the difference between `className?: string | undefined` and `className: string | undefined`.